webapp/src/index.js

```diff
@@ -27,8 +27,16 @@ class Plugin {
             const iconURL = getPluginURL(store.getState()) + '/public/app-bar-icon.png';
             registry.registerAppBarComponent(
                 iconURL,
-                (channel) => {
-                    startMeeting(channel.id)(store.dispatch, store.getState);
+                async (channel) => {
+                    if (channel) {
+                        startMeeting(channel.id, '')(store.dispatch, store.getState);
+                    } else {
+                        const teamId = store.getState().entities.teams.currentTeamId;
+                        const threadId = store.getState().views.threads.selectedThreadIdInTeam[teamId];
```

Comment:
What will be the value of threadId, if we are accessing it inside a channel?

Reply:
If there is a thread open in RHS, it gives the threadId of that thread and if no thread is present in rhs, it gives an empty string

Reply:
Note that we don't want to change the behavior when we are in center channel view. When a thread is open in the RHS, we should still post in the center channel. The scope of this change should only effect the "Threads View"

Reply:
@mickmister The scope of this change only affects the "Threads View", there is no change in the behavior when we are in the center channel view.

Reply:
@mickmister The scope of this change only affects the "Threads view". There is no change in the behavior when we are in the center channel view.

Reply:
@mickmister The scope of this change only affects the "Threads view". There is no change in the behavior when we are in the center channel view.